service/src/java/org/apache/hive/service/Service.java

```diff
@@ -63,6 +68,15 @@ public enum STATE {
    */
   void start();
 
+  /**
+   * Shut down the service properly before stopping
```

Comment:
can you add more detailed comment on what happens when this method is called and how it works?

Reply:
Sure. Thank you for the review!